README.md

```diff
@@ -38,65 +37,125 @@ python manage.py createsuperuser
 
 ## Generate the corpus and upload the data into elastic search
 
-1. Generate the xml from the json
-```bash
-docker-compose up corpus-creator-covid
-```
+1. Generate the XML article files from the JSON file:
 
-2. Upload the data into ES using logstah
-```bash
-docker-compose up logstash-covid
-```
+    ```bash
+    docker-compose up corpus-creator
+    ```
 
-## Start the api server
+2. Upload the data into ES using Logstash:
+
+    ```bash
+    docker-compose up logstash
+    ```
+
+## Start the `api` server
+
+Start the `api` container in Docker Compose:
 
-1. Start the qa container in docker
 ```bash
-docker-compose up -d api-covid
+docker-compose up api
 ```
 
-The swagger description of the services is in http://localhost:5000/
-
+The Swagger description of the services is in http://localhost:8000/
 
 ## Interact with the model
 
-The qa-covid container has a `main.py` script with some commands useful to test the model. 
+The `qa` container has a `main.py` script with some commands useful to test the model. 
 All the commands allow the flag `--ignore-es` to work with the testing data.
 
-1. Go into the `qa-covid` container.
-If the container is runnig
-```bash
-docker-compose exec qa-covid bash
-```
+1. Go into the `qa` container.
 
-If the containar is not running
-```bash
-docker-compose run qa-covid bash
-```
+    If the container is running:
 
-2. Inside the container, activate the conda env.
-```bash
-conda activate covid19-qa
-```
+    ```bash
+    docker-compose exec qa bash
+    ```
+    
+    If the container isn't running:
+
+    ```bash
+    docker-compose run qa bash
+    ```
+
+2. Inside the container, activate the conda env:
+
+    ```bash
+    conda activate covid19-qa
+    ```
 
 3. Execute the command. (Check out the help (`./main.py -h`) to see the available options.)
 Some examples:
+
+    ```bash
+    # Execute the `try` with some optimizations:
+    ./main.py --batch-size 672 --device 0 --threads 20
+    # Execute the `try` with some optimizations and without Elasticsearch:
+    ./main.py --batch-size 672 --device 0 --threads 20 --ignore-es
+    # Execute the interactive mode:
+    ./main.py interact
+    # Execute the interactive mode and without Elastichsearch:
+    ./main.py --ignore-es interact
+    ```
+
+## Query over Elasticsearch
+
+A useful tool to interact with your elastic search cluster is [Kibana](https://www.elastic.co/kibana).
+
+1. Run the Kibana container:
+
+    ```bash
+    docker-compose up kibana
+    ```
+
+2. In your browser, go to http://0.0.0.0:15601/app/kibana
+
+## Make Docker work with Nvidia
+
+The `qa` image is CUDA-enabled. It needs to run with the `nvidia` runtime to work well.
+
+1. Install [nvidia-docker-runtime](https://github.com/NVIDIA/nvidia-container-runtime).
```

Comment:
For just Docker, this is enough. But not for Docker Compose, unfortunately. For it, we need the next step below.